File: support/mock_backend/wiringPi/wiringPiI2C.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "wiringPi/wiringPiI2C.h"
int wiringPiI2CRead(int fd) { (void)fd;return 1; }
int wiringPiI2CReadReg8(int fd, int reg) { (void)fd;(void)reg;return 1; }
int wiringPiI2CReadReg16(int fd, int reg) { (void)fd;(void)reg;return 1; }
int wiringPiI2CWrite(int fd, int data) { (void)fd;(void)data;return 1; }
int wiringPiI2CWriteReg8(int fd, int reg, int data) { (void)fd;(void)reg;(void)data;return 1; }
int wiringPiI2CWriteReg16(int fd, int reg, int data) { (void)fd;(void)reg;(void)data;return 1; }
int wiringPiI2CSetupInterface(const char * device, int devId) { (void)device;(void)devId;return 1; }
int wiringPiI2CSetup(const int devId) { (void)devId;return 1; }

File: tests/wiringPi/wiringPiI2C.py
import importlib
import pytest

m=importlib.import_module("wiringop.wiring_pi.wiring_pi_i2c")

def test_wiring_pi_i2c_read():
    _=m.wiring_pi_i2c_read(1)

def test_wiring_pi_i2c_read_reg8():
    _=m.wiring_pi_i2c_read_reg8(1, 1)

def test_wiring_pi_i2c_read_reg16():
    _=m.wiring_pi_i2c_read_reg16(1, 1)

def test_wiring_pi_i2c_write():
    _=m.wiring_pi_i2c_write(1, 1)

def test_wiring_pi_i2c_write_reg8():
    _=m.wiring_pi_i2c_write_reg8(1, 1, 1)

def test_wiring_pi_i2c_write_reg16():
    _=m.wiring_pi_i2c_write_reg16(1, 1, 1)

def test_wiring_pi_i2c_setup_interface():
    _=m.wiring_pi_i2c_setup_interface("x", 1)

def test_wiring_pi_i2c_setup():
    _=m.wiring_pi_i2c_setup(1)
